hilbert/scheduler.py

```diff
@@ -31,23 +29,78 @@ def step(self):
             self.pointer += 1
 
 
-class LinearLRScheduler:
+class LearningRateScheduler:
+    def __init__(self,
+                 optimizer,
+                 start_lr,
+                 num_epochs,
+                 end_lr=0):
+        """
+        Base class for all learning rate schedulers.
+
+        :param optimizer: Optimizer used for training objective.
+        :param start_lr: Initial learning rate.
+        :param num_epochs: Number of epochs
+        :param end_lr: Stationary state learning rate
+        """
 
-    def __init__(self, optimizer, start_lr, num_epochs, end_lr=0):
         self.opt = optimizer
         self.start_lr = start_lr
         self.end_lr = end_lr
         self.num_epochs = num_epochs
         self.cur_epoch = -1
         self.step()
 
+    def step(self):
+        raise NotImplementedError('Not implemented!')
+
+
+class LinearLRScheduler(LearningRateScheduler):
+
+    def __init__(self, optimizer, start_lr, num_epochs, end_lr=0):
+        '''
+
+        :param optimizer: Optimizer used for training objective.
+        :param start_lr: Initial learning rate.
+        :param num_epochs: Number of epochs
+        :param end_lr: Stationary state learning rate
+        '''
+        super(LinearLRScheduler, self).__init__(optimizer,
```

Comment:
Minimal indent style
```suggestion
        super(LinearLRScheduler, self).__init__(
            optimizer,
```